### PR TITLE
VfsPath::join(): treat '..' at root as valid

### PR DESCRIPTION
**Pull request by Property404**
In unix systems, trying to access the parent of root will return root. This commit implements that behavior.

### CODE REVIEW

**Comment by manuel-woelker** [comment]
Thanks for the PR, although I am not quite sure this would be the expected behaviour. AFAICT the parent of the root being the root is a result of making the implementation of paths simpler, by enforcing every path having a parent.

This change would cause internal inconsistencies since currently `root.parent() == None`.

Navigating to ".." from root is probably an error or at least not intended. I'd rather have a proper Error when resolving such paths. To better understand the use case, do you have any concrete examples where this behaviour might be useful?

**Comment by Property404** [comment]
My use case is that I use VFS to emulate a unix-like system, and the current behavior causes unexpected results  

Quote from Stackoverflow:
"While it may seem seem more logical to throw an error, this behavior allows all directories to be treated the same rather than having to treat the root directory as a special case."

And I think that makes sense, at least for what I'm doing. But I'm not sure what other people are using this crate for

**Comment by Property404** [comment]
Added a `fixup!` for the `VfsPath::parent()` issue

File: src/path.rs [hunk 1]
```diff
@@ -566,17 +553,19 @@ impl VfsPath {
     /// use vfs::{MemoryFS, VfsError, VfsFileType, VfsMetadata, VfsPath};
```
Comment (manuel-woelker) [inline src/path.rs:553]: The doc comments above (i.e. "Returns `None` if this is a root path") should probaby be updated as well.

**Review by manuel-woelker — COMMENTED** [review]
Thanks for the fixup! LGTM, just a small doc change request.

**Comment by Property404** [comment]
Fixed doc comment and squashed/rebased

**Pull request merged**